recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -24,47 +26,59 @@ class OpenTelemetryCppConan(ConanFile):
         "fPIC": True,
         "shared": False,
     }
-    generators = "cmake", "cmake_find_package_multi"
     short_paths = True
 
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("abseil/20211102.0")
-        self.requires("grpc/1.45.2")
-        self.requires("libcurl/7.83.1")
-        self.requires("nlohmann_json/3.10.5")
-        self.requires("openssl/1.1.1o")
-        self.requires("opentelemetry-proto/0.18.0")
-        self.requires("protobuf/3.21.1")
-        self.requires("thrift/0.15.0")
-        if tools.Version(self.version) >= "1.3.0":
-            self.requires("boost/1.79.0")
+        self.requires("abseil/20220623.0")
+        self.requires("grpc/1.48.0")
+        self.requires("libcurl/7.85.0")
+        self.requires("nlohmann_json/3.11.2")
+        self.requires("openssl/1.1.1q")
+        if Version(self.version) <= "1.4.1":
+            self.requires("opentelemetry-proto/0.11.0")
+        else:
+            self.requires("opentelemetry-proto/0.19.0")
+        self.requires("protobuf/3.21.4")
+        self.requires("thrift/0.17.0")
+        if Version(self.version) >= "1.3.0":
+            self.requires("boost/1.80.0")
 
     def validate(self):
-        if self.settings.arch != "x86_64":
-            raise ConanInvalidConfiguration("Architecture not supported")
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+
+        if self.info.settings.arch != "x86_64":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support architecture : {self.info.settings.arch}")
 
-        if (self.settings.compiler == "Visual Studio" and
-           tools.Version(self.settings.compiler.version) < "16"):
-            raise ConanInvalidConfiguration("Visual Studio 2019 or higher required")
+        if (self.info.settings.compiler == "Visual Studio" and Version(self.info.settings.compiler.version) < "16") or \
```

Comment:
check min vs helper?

Reply:
Thank you for your review!
It is very useful helper.

Fixed.